qiling/utils.py

```diff
@@ -312,18 +312,22 @@ def ql_setup_logging_stream(ql, logger=None):
     ql_mode = ql.output
 
     # setup StreamHandler for logging to stdout
-    #ch = logging.StreamHandler()
-    ch = logging.getLogger()
+    if ql.log_console == True:
+        ch = logging.StreamHandler()
+    else:
+        # not print out to stdout by using NullHandler
+        ch = logging.NullHandler()
+
     ch.setLevel(logging.DEBUG)
     
     """
     guess this was added due to the incompatablity
     with print ("", end ="") with logger, 
     fix it with concat, will keep it for now
     """
-    # if ql_mode in (QL_OUT_DISASM, QL_OUT_DUMP):
-    #     # use empty string for newline if disasm or dump mode was enabled
-    #     ch.terminator = ""
+    if ql_mode in (QL_OUT_DISASM, QL_OUT_DUMP):
+        # use empty string for newline if disasm or dump mode was enabled
+        ch.terminator = ""
```

Comment:
i think this we gonna have issue in dprint. it will take out all the default \n ?

Reply:
if we use it with format string like print("%s" % var) will not cause any trouble, and i test these two modes with hello_world example its good so far.

Reply:
Any specific reason why we need this ? Since we already fix the issue in nprint

Reply:
since we are using the python logging module, its not supporting keyword "end"  like normal print did, and this is the most easy way to do it, set all stream with empty string terminated and print it out with append newline character or the specific one assigned by keyword "end" we need.

Reply:
i mean only in QL_OUT_DISASM, QL_OUT_DUMP why ?

Reply:
its special needs for `ql_hook_code_disasm`, or I can move it to nprint function, change the terminator while using keyword end , and huge context switch will be expected due to these two modes use `ql_hook_code_disasm` a lot.
So I think keep it empty default and add a newline character when we need it at nprint will be more efficiency.